### ISSUE TITLE
Introduce the new Navigation editor to the world

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#24683]
### Introduction

Currently as part of Gutenberg's phase 2 process there is an [effort](https://github.com/WordPress/gutenberg/projects/31) put into upgrading the current menus screen to a block based editor. With the latest progress in #24503 the project is in an architecturally stable point.

It is a good moment to remove the experimental cloak and have it available for a larger user base via the Gutenberg plugin. Issues such as ensuring the best possible backwards compatibility ( #23739 ) need more participation. Also the UI and UX explorations will advance smoother with more feedback.

Then, in the near future, after the Navigation editor is out of being just an experiment, it becomes beta software and the project's focus will be on fixing it, including the default workflow it will offer, and therefore move on with replacing the current screens. Also, aside from revamping the UX of making menus, this project allows users to bring blocks into their menus and that requires themes to opt in. Testing this important feature via the Gutenberg plugin sooner rather than later is essential for it to work properly.

### Actions

- [ ] Remove the experimental flags that hide the Navigation editor
- [ ] Check the filter to opt out works as expected
- [ ] Release the Navigation editor as a default Appearance > Menus link in Gutenberg

**Comment by aristath** [comment]
@StaggerLeee in time it will be possible to use some more blocks inside menus so it will be possible to build what most people refer to as "mega menu". But that is not the current focus, or the point of this ticket :+1:

**Comment by noisysocks** [comment]
Before a developer can pick up this issue, we need to know which design changes are required for us to be happy with soft launching the Navigation screen. Do you have a list, @shaunandrews?

**Comment by shaunandrews** [comment]
@noisysocks I outlined my thoughts over here: https://github.com/WordPress/gutenberg/issues/24875

**Comment by noisysocks** [comment]
Thanks @shaunandrews!

So, to summarise the status here, we're blocked on #24875 and #24760. One those two overview issues are addressed we can remove the experimental flag.

**Comment by sirreal** [comment]
Has this issue been superseded by another or should we point to another place for tracking? The information here is stale to the best of my knowledge.

**Comment by noisysocks** [comment]
Hey @sirreal. The best way to track the status of this project is by checking the [project board](https://github.com/WordPress/gutenberg/projects/31). Currently the contributors that were working full time on this project have hit pause and are focusing instead on Widgets. It's likely we'll come back to this for WordPress 5.7.

**Comment by draganescu** [comment]
Moving this from blocked as both blocking issues are now closed.

**Comment by getdave** [comment]
As we're not yet ready to do this I've moved to the "Medium" priority section of the board. This reduces visual "clutter" and helps clear up what is Highest Priority,

**Comment by draganescu** [comment]
Wow accidentally made an issue of a task here :D

**Comment by skorasaurus** [comment]
> Wow accidentally made an issue of a task here :D

Might be a result of the [changes to Github issues](https://wptavern.com/major-revamp-coming-to-github-issues) some of which have already been released? :)

**Comment by superbmario** [comment]
Regarding "Check the filter to opt out works as expected" - what will the filter be named? I'd like to start adding that to sites I know won't be needing this feature to avoid possible issues. Knowing the name in advance will be super useful for people like me, so if it not yet determined then I'd like to suggest it be created soon.

**Comment by getdave** [comment]
> Regarding "Check the filter to opt out works as expected" - what will the filter be named? I'd like to start adding that to sites I know won't be needing this feature to avoid possible issues. Knowing the name in advance will be super useful for people like me, so if it not yet determined then I'd like to suggest it be created soon.

This is the format used by the Widgets Editor

```php
// Disables the block editor from managing widgets in the Gutenberg plugin.
add_filter( 'gutenberg_use_widgets_block_editor', '__return_false' );
// Disables the block editor from managing widgets.
add_filter( 'use_widgets_block_editor', '__return_false' );
```

Therefore we should _probably_ follow the same format, namely:

```php
// Disables the block editor from managing navigations in the Gutenberg plugin.
add_filter( 'gutenberg_use_navigation_block_editor', '__return_false' );
// Disables the block editor from managing navigations.
add_filter( 'use_navigation_block_editor', '__return_false' );
```

⚠️ We're not _guaranteeing_ this at this point however. The opt in/out strategy might need to be different for the Nav Editor.

**Comment by superbmario** [comment]
I was wondering if the filters to disable the new Navigations editor are added yet. I cannot seem to find it in the source. I've opened a new issue to track just in case https://github.com/WordPress/gutenberg/issues/37640

Thanks!

**Comment by draganescu** [comment]
I will close this issue as it is referring to some really outdated information about the Navigation editor.
